components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java

```diff
@@ -76,6 +74,11 @@ public void startRoute(String id) throws Exception {
         camelContext.getRouteController().startRoute(id);
     }
 
+    public void resetRoute(String id) throws Exception {
+        ManagedRouteMBean resetRoute = camelContext.getManagedRoute(id, ManagedRouteMBean.class);
+        resetRoute.reset();
```

Comment:
HTTP 404 is definitely the correct status code here.  This indicates a client error, i.e. the client provided an incorrect route ID (an incorrect resource URI) which does not exist.  HTTP 204 has different semantics and is not appropriate here:  It would mean that the request succeeded, but the server response does not contain any body (often used as the response code for DELETE operations).